core/src/test/scala/unit/kafka/server/ListOffsetsRequestTest.scala

```diff
@@ -192,7 +192,7 @@ class ListOffsetsRequestTest extends BaseRequestTest {
     killBroker(firstLeaderId)
     val secondLeaderId = TestUtils.awaitLeaderChange(servers, partition, firstLeaderId)
     // make sure high watermark of new leader has caught up
-    TestUtils.waitUntilTrue(() => sendRequest(secondLeaderId, 0L, -1).errorCode() != Errors.OFFSET_NOT_AVAILABLE.code(),
+    TestUtils.waitUntilTrue(() => sendRequest(secondLeaderId, ListOffsetsRequest.LATEST_TIMESTAMP, -1).errorCode != Errors.OFFSET_NOT_AVAILABLE.code,
```

Comment:
`OFFSET_NOT_AVAILABLE` is only returned when `LATEST_TIMESTAMP` or `MAX_TIMESTAMP` are used.

Reply:
So, if my understanding is correct, the test failure is because we are using the the `OFFSET_NOT_AVAILABLE` to identify if the new leader has caught up, but the `OFFSET_NOT_AVAILABLE` error only throws when requesting `LATEST_TIMESTAMP` or a timestamp beyond the last fetchable offset. Since we're using a small number of timestamp, this wait will pass immediately, and cause the below assertion failure with `OFFSET_NOT_AVAILABLE` sometimes. 

Is this correct? If so, this change LGTM! Thanks.

Reply:
Yep. This is correct.

Reply:
Thanks for fixing this!